src/core/a-mixin.js

```diff
@@ -9,106 +9,97 @@ var MULTIPLE_COMPONENT_DELIMITER = '__';
-        this.cacheAttribute(attr, value);
-      }
-    },
+  connectedCallback () {
+    this.id = this.getAttribute('id');
+    this.sceneEl = this.closestScene();
+    this.cacheAttributes();
+    this.updateEntities();
+    this.load();
+  }
 
-    /**
-     * If `attr` is a component, then parse the value using the schema and store it.
-     */
-    cacheAttribute: {
-      value: function (attr, value) {
-        var component;
-        var componentName;
+  /**
+   * setAttribute that parses and caches component values.
+   */
+  setAttribute (attr, value) {
+    window.HTMLElement.prototype.setAttribute.call(this, attr, value);
+    this.cacheAttribute(attr, value);
+  }
 
-        // Get component data.
-        componentName = utils.split(attr, MULTIPLE_COMPONENT_DELIMITER)[0];
-        component = components[componentName];
-        if (!component) { return; }
-        if (value === undefined) {
-          value = window.HTMLElement.prototype.getAttribute.call(this, attr);
-        }
-        this.componentCache[attr] = component.parseAttrValueForCache(value);
-      }
-    },
+  /**
+   * If `attr` is a component, then parse the value using the schema and store it.
+   */
+  cacheAttribute (attr, value) {
+    var component;
+    var componentName;
 
-    /**
-     * If `attr` is a component, then grab pre-parsed value from the cache.
-     * Else do a normal getAttribute.
-     */
-    getAttribute: {
-      value: function (attr) {
-        return this.componentCache[attr] ||
-               window.HTMLElement.prototype.getAttribute.call(this, attr);
-      }
-    },
+    // Get component data.
+    componentName = utils.split(attr, MULTIPLE_COMPONENT_DELIMITER)[0];
+    component = components[componentName];
+    if (!component) { return; }
+    if (value === undefined) {
+      value = window.HTMLElement.prototype.getAttribute.call(this, attr);
+    }
+    this.componentCache[attr] = component.parseAttrValueForCache(value);
+  }
+
+  /**
+   * If `attr` is a component, then grab pre-parsed value from the cache.
+   * Else do a normal getAttribute.
+   */
+  getAttribute (attr) {
+    return this.componentCache[attr] ||
+      window.HTMLElement.prototype.getAttribute.call(this, attr);
+  }
 
-    /**
-     * Parse and cache every component defined on the mixin.
-     */
-    cacheAttributes: {
-      value: function () {
-        var attributes = this.attributes;
-        var attrName;
-        var i;
-        for (i = 0; i < attributes.length; i++) {
-          attrName = attributes[i].name;
-          this.cacheAttribute(attrName);
-        }
-      }
-    },
+  /**
+   * Parse and cache every component defined on the mixin.
+   */
+  cacheAttributes () {
+    var attributes = this.attributes;
+    var attrName;
+    var i;
+    for (i = 0; i < attributes.length; i++) {
+      attrName = attributes[i].name;
+      this.cacheAttribute(attrName);
+    }
+  }
```

Comment:
following Justin hints, while it's not necessary to refactor everything at this point, an example on how above method could be improved is:

```js
const {attributes} = this;
for (let i = 0; i < attributes.length; i++) {
  const {name} = this.attributes[i]; // or {name, value}
  this.cacheAttribute(name);
}
```

Reply:
thanks, as mentioned above, A-Frame code base is ES5 prior to this PR. We can revisit and refactor later.